mesonbuild/dependencies/base.py

```diff
@@ -2370,6 +2370,10 @@ def find_external_dependency(name, env, kwargs):
         except DependencyException as e:
             pkg_exc.append(e)
             mlog.debug(str(e))
+        except MesonException as e:
+            if 'non-clang' in str(e):
+                continue
+            raise
```

Comment:
Wouldn't it be clearer to catch that in `AppleFrameworks:__init__` and turn it into `is_found = False`, as `ExtraFrameworkDependency` does?

Reply:
Updated.